pkg/alertmanager/operator.go

```diff
@@ -904,6 +904,33 @@ func (c *Operator) selectAlertManagerConfigs(ctx context.Context, am *monitoring
 			if err != nil {
 				break
 			}
+
+			for j, whConfig := range receiver.WebhookConfigs {
```

Comment:
we need to validate that `whConfig.URL` and `whConfig.URLSecret` aren't both `nil`. In case of `whConfig.URLSecret`, we need to verify that the referenced key exists.

Reply:
I'm checking it [here](https://github.com/prometheus-operator/prometheus-operator/pull/3601/files#diff-a66b591453b9fc72f73c1c29df8f7448add564f190f857a44c54967fde4ea6c6R347)

Reply:
it would be better to do it here as the operator will skip the invalid `AlertmanagerConfig` instead of failing the whole  reconciliation.

Reply:
Moved